front/src/applications/operationalStudies/views/v2/ManageTrainScheduleV2.tsx

```diff
@@ -89,22 +76,22 @@ const ManageTrainScheduleV2 = () => {
         <img src={pahtFindingPic} alt="path finding" />
         <span className="ml-2 d-flex align-items-center flex-grow-1 w-100">
           {t('tabs.pathFinding')}
-          {pathFinding?.length && !Number.isNaN(pathFinding.length) && (
+          {destination && destination.positionOnPath && (
             <small className="ml-auto pl-1">
-              {pathFinding.length && formatKmValue(pathFinding.length / 1000, 3)}
+              {formatKmValue(destination.positionOnPath / 1000000, 3)}
```

Comment:
Can you modify this function ?
- [ ] it should not take value potentially undefined
- [ ] set the default digits value to 3
- [ ] add a new param which indicates the unit of value ('meter' by default). If it is 'meter', then divide value by 1000. If value is 'millimeter', then divide it by 1000000
```suggestion
              {formatKmValue(destination.positionOnPath!, 'millimeters')}
```

Reply:
Done